src/test/java/uk/gov/hmcts/reform/pip/account/management/service/AccountServiceTest.java

```diff
@@ -685,4 +691,180 @@ void testUpdateAccountUnexpectedDateTimeFormat() {
+        }, "The exception when a user has not been found has been thrown");
+
+        assertTrue(notFoundException.getMessage().contains(userId.toString()),
+                   "Exception message thrown does not contain the user ID");
+    }
+
+    @Test
+    void testFindAllAccountsExceptThirdPartyEmptyParams() {
+        Pageable pageable = PageRequest.of(0, 25);
+        List<UserProvenances> emptyUserProvenancesList = new ArrayList<>();
+        List<Roles> emptyRoleList = new ArrayList<>();
+        Page<PiUser> page = new PageImpl<>(List.of(piUser), pageable, List.of(piUser).size());
+        when(userRepository.findAllByEmailLikeAndUserProvenanceInAndRolesInAndProvenanceUserIdLike(
+            argThat(arg -> "%%".equals(arg)),
+            any(),
+            any(),
+            any(),
+            any()
+        )).thenReturn(page);
+
+        Page<PiUser> response = accountService.findAllAccountsExceptThirdParty(pageable, "", "",
+                                                                               emptyUserProvenancesList, emptyRoleList,
+                                                                               "");
+
+        assertEquals(page, response, "Returned page did not match expected");
+
+    }
+
+    @Test
+    void testFindAllAccountsExceptThirdPartyFullParams() {
+        Pageable pageable = PageRequest.of(0, 25);
+        List<UserProvenances> userProvenancesList = List.of(UserProvenances.PI_AAD);
+        List<Roles> roleList = List.of(Roles.VERIFIED);
+        Page<PiUser> page = new PageImpl<>(List.of(piUser), pageable, List.of(piUser).size());
+        when(userRepository.findAllByEmailLikeAndUserProvenanceInAndRolesInAndProvenanceUserIdLike(
+            any(),
+            any(),
+            any(),
+            any(),
+            any()
+        )).thenReturn(page);
+
+        Page<PiUser> response = accountService.findAllAccountsExceptThirdParty(pageable, "test", ID,
+                                                                               userProvenancesList, roleList, "");
+
+        assertEquals(page, response, "Returned page did not match expected");
+
+    }
+
+    @Test
+    void testFindAllAccountsExceptThirdPartyOnlyUserId() {
+        Pageable pageable = PageRequest.of(0, 25);
+        PiUser user = new PiUser();
+        user.setUserId(UUID.randomUUID());
+        List<UserProvenances> emptyUserProvenancesList = new ArrayList<>();
+        List<Roles> emptyRoleList = new ArrayList<>();
+        Page<PiUser> page = new PageImpl<>(List.of(user), pageable, List.of(user).size());
+
+        when(userRepository.findByUserIdPageable(user.getUserId().toString(), pageable)).thenReturn(page);
+
+        Page<PiUser> response = accountService.findAllAccountsExceptThirdParty(pageable, "", "",
+            emptyUserProvenancesList, emptyRoleList, user.getUserId().toString());
+
+        assertEquals(page, response, "Returned page did not match expected");
+    }
```

Comment:
As we return straight from this method after call to find user by ID. Could also verify the second call to the user repository never happen using:

verify(userRepository, never()).findAllByEmailLikeAndUserProvenanceInAndRolesInAndProvenanceUserIdLike(any(), any(), any(), any(), any());

Reply:
Done :) Have also updated the other tests with 

verify(userRepository, never()).findByUserIdPageable(any(), any());

When we are not searching by the user id to ensure its not called.